### PR TITLE
Fix `test_run_node` (rename feature `sandbox`)

### PR DESCRIPTION
**Pull request by yvan-sraka**

### CODE REVIEW

**Review by AureliaDolo — DISMISSED** [review]

**Comment by yvan-sraka** [comment]
bors merge

**Comment by yvan-sraka** [comment]
bors merge

**Comment by yvan-sraka** [comment]
bors merge

**Comment by AureliaDolo** [comment]
It's that test that is failing again

**Comment by yvan-sraka** [comment]
> It's that test that is failing again

Yes! I don't why, and I hesitate to disable it in one PR and to fix it in another to unstuck merging?!

**Comment by AureliaDolo** [comment]
yaeh that may be the quick patch we need :/

**Comment by yvan-sraka** [comment]
> yaeh that may be the quick patch we need :/

:heavy_check_mark:

**Review by AureliaDolo — COMMENTED** [review]
There are clippy warnings, and I'm not sure I understand the branch you are based upon

**Review by AureliaDolo — CHANGES_REQUESTED** [review]
There is too much stuff not related to this PR description in this PR

**Comment by yvan-sraka** [comment]
> There is too much stuff not related to this PR description in this PR

So that's why I first split it into https://github.com/massalabs/massa/pull/2310 and will turn this PR to draft! Reviews and comment are still welcomed 🙂

**Comment by AureliaDolo** [comment]
> > There is too much stuff not related to this PR description in this PR
> 
> So that's why I first split it into #2310 and will turn this PR to draft! Reviews and comment are still welcomed slightly_smiling_face

aaah ok thanks :)

Maybe you could edit the description/title, even if the other PR is not merged yet  ?

**Comment by AurelienFT** [comment]
Can we add description on the purpose of this PR and resolve conflicts ? So that we can make a review.

**Comment by yvan-sraka** [comment]
> Can we add description on the purpose of this PR and resolve conflicts ? So that we can make a review.

Conflict resolved! This PR re-enables a disabled end-to-end test scenario that try to launch and then stop a node gracefully, the idea is to extend this test as described here https://github.com/massalabs/massa/issues/1134#issuecomment-1032662328

**Review by AurelienFT — CHANGES_REQUESTED** [review]
I tried your PR and when trying to launch the client I get the following error : 
```
thread 'main' panicked at 'failed to read base_config/config.toml config /Users/aurelien/tmp/massa/massa-client: configuration file "base_config/config.toml" not found', /Users/aurelien/tmp/massa/massa-models/src/node_configuration/massa_settings.rs:42:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

File: massa-node/src/tests/config.toml [hunk 1]
```diff
@@ -1,95 +0,0 @@
-[logging]
```
Comment (AurelienFT) [inline massa-node/src/tests/config.toml:1]: Why deleting this file ?

File: massa-client/src/settings.rs [hunk 2]
```diff
@@ -1,28 +1,54 @@
...
+
+#[cfg(not(feature = "testing"))]
 lazy_static::lazy_static! {
     pub static ref SETTINGS: Settings = build_massa_settings("massa-client", "MASSA_CLIENT");
 }
 
 #[derive(Debug, Deserialize, Clone)]
 pub struct Settings {
-    pub default_node: DefaultNode,
+    pub default_node: NodeAddr,
```
Comment (AurelienFT) [inline massa-client/src/settings.rs:22]: This change cause this error : 
```rust
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: missing field `public`', /Users/aurelienfoucault/massa/massa-models/src/node_configuration/massa_settings.rs:78:25
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I think you should change the format in the config file.

File: massa-client/src/tests/scenarios.rs [hunk 3]
```diff
@@ -6,13 +6,10 @@ use serde::de::DeserializeOwned;
 use serial_test::serial;
 use std::{thread::JoinHandle, time::Duration};
 
-const TIMEOUT: u64 = 30;
```
Comment (AurelienFT) [inline massa-client/src/tests/scenarios.rs:9]: Why removing this ? It can't be useful for further tests ?

**Review by AurelienFT — CHANGES_REQUESTED** [review]

File: massa-node/src/tests/config.toml (hunk 1 above)
Comment (yvan-sraka) [reply massa-node/src/tests/config.toml:1]: It's no longer used, I guess …

File: massa-client/src/settings.rs (hunk 2 above)
Comment (yvan-sraka) [reply massa-client/src/settings.rs:22]: You're right!

File: massa-client/src/tests/scenarios.rs (hunk 3 above)
Comment (yvan-sraka) [reply massa-client/src/tests/scenarios.rs:9]: True :)

**Comment by AurelienFT** [comment]
The node is not running well and it's hard to debug the command. I let it for now but still need some fixes to do.

**Comment by damip** [comment]
Closing this for now

**Pull request closed without merging**